src/main/kotlin/step2/calculator/Expression.kt

```diff
@@ -0,0 +1,25 @@
+package step2.calculator
+
+class Expression(
+    private val prev: Double,
+    private val operator: Operator,
+    private val next: Double
+) {
+    fun calculate(): Double {
+        validateExpressionAvailability(prev, operator, next)
+
+        return operator.calculate(prev, next)
+    }
+
+    private fun validateExpressionAvailability(
+        @Suppress("UNUSED_PARAMETER") prev: Double,
+        operator: Operator,
+        next: Double
+    ) {
+        when (operator) {
+            Operator.DIVISION -> next.takeIf { next > 0 }
+                ?: throw IllegalArgumentException("It cannot be divided by zero.")
+            else -> println("Computable formula.")
```

Comment:
`takeIf` 확장함수 사용 👍 👍 👍 

개인적으로는 그냥 if문을 사용하는게 더 가독성이 좋지 않았을까 싶어요. 🤔  
학습 차원에서 when 구문이나 takeIf를 사용하신거라면 그대로 두셔도 무방합니다.✨✨  😃👍

```suggestion
            if(next > 0) {
                throw IllegalArgumentException("It cannot be divided by zero.")
            } else { 
                println("Computable formula.")
            }
```

또한, 위 식은 아래와 같이 `require`로 더 깔끔하게 바꿔볼 수 있을 것 같아요. 🙌
```suggestion
            Operator.DIVISION -> require(next > 0) { "It cannot be divided by zero." }
            else -> println("Computable formula.")
```


추가적으로 한 가지 질문이 있습니다! 🙋‍♂️
print문은 로깅을 남긴다고 이해했는데 맞을까요? 🤔

Reply:
` ./gradlew clean check` 을 실행시켰을 때 아래와 같은 내용이 발생하여, else문에 로깅을 추가하게 되었습니다.

`Non exhaustive 'when' statements on enum will be prohibited in 1.7, add 'PLUS', 'MINUS', 'MULTIPLE' branches or 'else' branch instead`

Reply:
require를 사용하는 방법으로 변경하였습니다!! 가독성이 전보다 더 나아진거 같아요 🙇